src/net.cpp

```diff
@@ -2633,8 +2637,10 @@ CNode::CNode(NodeId idIn, ServiceFlags nLocalServicesIn, int nMyStartingHeightIn
     hSocket = hSocketIn;
     addrName = addrNameIn == "" ? addr.ToStringIPPort() : addrNameIn;
     hashContinue = uint256();
-    m_tx_relay = MakeUnique<TxRelay>();
-    m_tx_relay->pfilter = MakeUnique<CBloomFilter>();
+    if (!blocks_only) {
+        m_tx_relay = MakeUnique<TxRelay>();
+        m_tx_relay->pfilter = MakeUnique<CBloomFilter>();
```

Comment:
Nit: this initialization of the `TxRelay::pfilter` field probably belongs better in the `TxRelay` constructor.

Reply:
Done.